docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,142 @@
+# [ZeroConf](https://en.wikipedia.org/wiki/Zero-configuration_networking) Protocol Implementation
+
+## Goal
+
+Agent implements [Zero-configuration networking](https://en.wikipedia.org/wiki/Zero-configuration_networking) (hence 'ZeroConf'), a set of technologies that help discover devices and services using DNS-based discovery. There are 2 main elements: Multicast DNS (mDNS) and DNS-based Service Discovery (DNS-SD).
+
+While ZeroConf is often used in home networks (that don't often include regular DNS), ZeroConf is broadly applicable and is useful in IoT deployments in which devices are transient, there are many devices, developers wish to dynamically manage services on these devices.
+
+These technologies require additional packages and shared libraries. Supporting ZeroConf as an Akri protcol possibly (!?) provides a mechanism by which (Kubernetes) application developers can leverage ZeroConf technologies without having to install or be familiar with ZeroConf dependencies.
+
+## Why ZeroConf?
+
+ZeroConf is a useful mechanism to publish services that have not only names (e.g. `device-123456`) but simple metadata (e.g. `_elevators._udp`) and limited textual data for e.g. labels. This permits scenarios where an Akri Broker implementation would want to query an e.g. building network to find its e.g. elevators and interact with these. The Broker would encapsulate the (proprietary protocol and) functionality needed to interact with the elevators and, thanks to Akri, would be able to expose these (elevators) as perhaps REST-based or gRPC-based services to Kubernetes applications.
+
+## Background
+
+For more information, see [Zero-configuration networking](https://en.wikipedia.org/wiki/Zero-configuration_networking).
```

Comment:
Maybe add a transitional sentence along the lines of: "To get a better understanding of Zeroconf, lets use Avahi, a Zeroconf implementation for Linux environments, to try out creating a service that is discoverable via Zeroconf. Then, we will discover the service and enumerate its attributes..."

Reply:
How about this? https://github.com/deislabs/akri/pull/162/commits/fe7859dc0dbdd7c73d7a315a8644baffdb4be398